src/sentry/eventstream/snuba.py

```diff
@@ -161,6 +161,21 @@ def insert(
         event_type = self._get_event_type(event)
         occurrence_data = self._get_occurrence_data(event)
 
+        # instead of normalizing and doing custom 'contexts' processing in snuba, we elect to do it here instead to
+        # avoid having to clutter up snuba code with business logic
+        if event_type == EventStreamEventType.Generic:
+            event_data = dict(event_data)
+            contexts = event_data.setdefault("contexts", {})
+
+            # add user.geo to contexts if it exists
+            user_dict = event_data.get("user") or event_data.get("sentry.interfaces.User") or {}
+            geo = user_dict.get("geo", {})
+            if "geo" not in contexts and isinstance(geo, dict):
+                contexts["geo"] = geo
+
+            # transactions processing has a configurable 'skipped contexts' to skip writing specific contexts maps
+            # to the row. for now, we're ignoring that until we have a need for it
+
```

Comment:
If we're going to process context data like this here, it might make sense to split this out into a separate function, and have a different function per event type

Reply:
Yeah, that's fair. I'll merge this in for now and refactor it into a separate function when we start adding more logic here.